2018-02-12  Zixuan Wang  <classkrn@lnxsrv09.seas.ucla.edu>

        * timespec.h: avoid the overflow by direct comparison instead of
        subtraction   fix the problems of incorrectly displayed time

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2018-02-12 05:22:41.024339000 -0800
@@ -45,8 +45,9 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec > b.tv_sec) {return 1;}
+  else if(a.tv_sec < b.tv_sec) {return -1;}
+  else {return 0;}
 }
 
 # if ! HAVE_DECL_NANOSLEEP
